src/BloomFilter/BloomFilterAbstract.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace RocketLabs\BloomFilter;
+
+use RocketLabs\BloomFilter\Hash\HashInterface;
+use RocketLabs\BloomFilter\Persist\PersisterInterface;
+
+/**
+ * @author Igor Veremchuk igor.veremchuk@rocket-internet.de
+ */
+abstract class BloomFilterAbstract implements BloomFilterInterface
+{
+    /** @var int */
+    protected $bitSize;
+    /** @var int */
+    protected $hashCount;
+    /** @var PersisterInterface */
+    protected $persister;
+    /** @var HashInterface */
+    protected $hash;
+
+    /**
+     * @param PersisterInterface $persister
+     * @param HashInterface $hash
+     * @param int $setSize
+     * @param float $falsePositiveProbability
+     *
+     * @return BloomFilterInterface
+     */
+    public static function createFromSetSize(
```

Comment:
In my opinion, would be more clear to have the factory method as abstract and each child class has to implement it... having to call the abstract class constructor and execute the code which is on the current init method.